### PR TITLE
Require numpy >=1.16

### PR DESCRIPTION
**Pull request by crusaderky**
Needed to work around a mamba bug in numpy 1.15 on MacOSX (#7227)

### CODE REVIEW

**Comment by jsignell** [comment]
This seems fine to me, especially given the conversation in https://github.com/dask/dask/issues/7378, but let's get a :+1: from @jrbourbeau as well.

File: setup.py [hunk 1]
```diff
@@ -8,9 +8,9 @@
 # NOTE: These are tested in `continuous_integration/test_imports.sh` If
 # you modify these, make sure to change the corresponding line there.
 extras_require = {
-    "array": ["numpy >= 1.15.1"],
+    "array": ["numpy >= 1.16"],
```
Comment (jsignell) [inline setup.py:11]: I'd prefer the mins to include the patch unless there is a reason not to. 

```suggestion
    "array": ["numpy >= 1.16.0"],
```

Comment (crusaderky) [reply setup.py:11]: See NEP-29: https://numpy.org/neps/nep-0029-deprecation_policy.html#detailed-description
I understand that it states (albeit in a less than perfectly clear wording) that support for patch versions would overdeliver on the contract stated by the NEP. The idea is that bumping a patch version is (typically) fairly painless.

From personal experience on xarray, pint, and my various personal projects, leaving the patch version free relieves quite a bit of pain from the maintainers while being hardly an issue for any user.

**Comment by jakirkham** [comment]
Any thoughts on just going to NumPy 1.17 as a minimum ( https://github.com/dask/dask/issues/7378#issuecomment-797613254 )? That still older than 18 months and will allow us to drop a lot of compatibility code

cc @pentschev (for vis)

**Comment by crusaderky** [comment]
@jakirkham I'm strongly in favour for bumping to 1.17 directly

**Review by jrbourbeau — COMMENTED** [review]
Thanks @crusaderky! Generally bumping to 1.16 seems okay to me

However I get the sense that this isn't super time sensitive (feel free to correct me if that's not the case). So given the discussion in https://github.com/dask/dask/issues/7378 and the appetite to bump to 1.17+, what do we think about waiting a few days to get input from the rest of the community on the broader minimum version policy and, if there's no significant pushback, we adopt the policy and update this PR to `numpy >= 1.17`?

**Review by jrbourbeau — APPROVED** [review]
Thanks for your work here @crusaderky! I'm going to merge this now as we're still coming to consensus over in https://github.com/dask/dask/issues/7378

**Pull request merged**